www/js/diary/current.js

```diff
@@ -0,0 +1,224 @@
+  $scope.diary = function() {
+    console.log('diary', "in diary");
+    $state.go("root.main.diary");
+  };   
+
+  angular.extend($scope.mapCtrl, {
+    defaults: {
+    }
+  });
+
+  angular.extend($scope.mapCtrl.defaults, Config.getMapTiles());
+
+  console.log("mapCtrl", $scope.mapCtrl);
+
+  var date = new Date();
+  var time = date.toTimeString().split(' ')[0].split(':');
+
+  var startTimeFn = function(ts) {
+    var date = new Date(ts*1000);
+    var hours = date.getHours();
+    var minutes = date.getMinutes();
+    var amOrPm = hours < 12 ? 'AM' : 'PM';
+    hours = hours % 12;
+    hours = hours ? hours : 12;
+    minutes = minutes < 10 ? '0'+ minutes : minutes;
+    $scope.startTime = hours + ':' + minutes + ' ' + amOrPm;
+  };
+
+  var msToMph = function(ms) {
+    var cmPerM = 100;
+    var inchPerCm = 1/2.54;
+    var footPerInch = 1/12;
+    var milePerFoot = 1/5280;
+    var secPerMin = 60;
+    var minPerHour = 60;
+    $scope.currSpeedInMph = Math.round(ms*cmPerM*inchPerCm*footPerInch*
+                            milePerFoot*secPerMin*minPerHour);
+  };
+
+  var degreeToDirection = function(degree) {
+    var index = Math.floor((degree / 45) + 0.5) % 8;
+    $scope.currentDirection = directions[index];
+  };
+
+  var refreshTrip = function() {
+    leafletData.getMap('current').then(function(map) {
+      return db.getAllSensorData(BACKGROUND_LOCATION);
+    }).then(function(result) {
+      console.log(result);
+      var coordinates = result.map(function(locWrapper, index, locList) {
+        return [locWrapper.data.longitude, locWrapper.data.latitude];
+      });
+      var both = [result[0].data.longitude, result[0].data.latitude];
+      var bearing = Math.round(result[0].data.bearing);
+      console.log(bearing);
+      startTimeFn(result[result.length - 1].data.ts);
+      msToMph(result[0].data.sensed_speed);
+      degreeToDirection(bearing);
+      angular.extend($scope.mapCtrl, { 
+        defaults : {
+          center: {
+            lat: both[1],
+            lng: both[0],
+            zoom: 15
+          }
+        },
+        markers: {
+          hereMarker: {
+            lat: both[1],
+            lng: both[0],
+            icon: {
+              iconUrl: 'img/test.gif',
+              iconSize: [20, 20],
+              iconAnchor: [10, 10],
+              popupAnchor: [0, 0],
+              shadowSize: [0, 0],
+              shadowAnchor: [0, 0]
+            },
+            iconAngle: bearing,
+          }
+        },
+        geojson: {
+          data: [
+            {
+              type: "LineString",
+              coordinates: coordinates
+            },
+          ],
+        },
+      });
+    });
+    console.log($scope.mapCtrl);
+  };
+
+  $scope.$on('leafletDirectiveMap.current.resize', function(event, data) {
+        console.log("current/map received resize event, invalidating map size");
+        data.leafletObject.invalidateSize();
+  });
+   
+  var addSafe = function(marker, ts, map) {
```

Comment:
Also these functions are internal function. Should I make them part of the factory?